### ISSUE TITLE
Post model: Stop scrubbing post properties (all of them?) prior to insert/update a WP_Post.

### ISSUE THREAD

**Issue by eri-trabiccolo** [gocodebox/lifterlms#1441]
Originally discussed here:

https://github.com/gocodebox/lifterlms/pull/1408#issuecomment-720696709

Basically `wp_insert_post()` should already sanitize the post data, see:
https://developer.wordpress.org/reference/functions/wp_insert_post/#security

**Comment by thomasplevy** [comment]
Agreed, I think that we just need to cover the abstract getters/setters with explicit tests and then we can see what effect we'll have with a change.
